### PR TITLE
Only master requests should be processed by the ResponseListener.

### PR DESCRIPTION
**Pull request by ManInTheBox**
Hi,

This PR should ensure that only master requests are handled by the `ResponseListener`. Sub-requests should be ignored.

I think that we need to avoid sub-requests because we only care about [master requests](https://github.com/ekino/EkinoNewRelicBundle/blob/master/Listener/RequestListener.php#L125) when starting NR transactions. That means that we must [end transaction](https://github.com/ekino/EkinoNewRelicBundle/blob/00f68a6bf836afeee9d83ac5e6b4087b89370b89/Listener/ResponseListener.php#L118) only if the request is a master one. If we process sub-requests in `ResponseListener` then we will accidentally, prematurely, end the master request transaction.

In other words, we always need to process _only_ master requests. This is already the case with `RequestListener` and now I fixed `ResponseListener` so no sub-requests can prematurely end our NR transactions.

Please let me know if I'm wrong. Thanks!

### CODE REVIEW

**Comment by Nyholm** [comment]
Im not sure I understand. Why wouldn't you want to log subrequests?

**Comment by ManInTheBox** [comment]
The bundle currently does not log sub-requests, it logs only [master](https://github.com/ekino/EkinoNewRelicBundle/blob/master/Listener/RequestListener.php#L125) ones. To demonstrate the problem this is the current flow what's happening (not providing logs, but could if necessary):

1. Master request fires and NR transaction starts
2. Subrequest fires and no new NR transaction is [started](https://github.com/ekino/EkinoNewRelicBundle/blob/master/Listener/RequestListener.php#L125)
3. Subrequest ends and NR transaction [ends](https://github.com/ekino/EkinoNewRelicBundle/blob/00f68a6bf836afeee9d83ac5e6b4087b89370b89/Listener/ResponseListener.php#L118) - this is wrong!
4. Master request ends but NR transaction was already ended in step 3!

Instead, we should do:

1. Master request fires and NR transaction starts
2. Subrequest fires and no new NR transaction is [started](https://github.com/ekino/EkinoNewRelicBundle/blob/master/Listener/RequestListener.php#L125)
3. Subrequest ends and **no** NR transaction ends
4. Master request ends and NR transaction ends now

The key point here is that only master request should be tracked. I'm not against tracking subrequests as well, but I think that's not currently supported by NR extension itself. I already tried that and realized that when `newrelic_end_transaction()` ends subrequest' transaction, it also means that you can't resume original master request transaction. That's the separate issue, outside the scope of this PR.

**Comment by Nyholm** [comment]
Thank you. I like this. 

It has been quite a few changes since you made this PR. Could you rebase on master and update your tests to PHPUnit 6?
